### ISSUE TITLE
Pallet-examples: Add missing hooks

### ISSUE THREAD

**Issue by ggwpez** [paritytech/polkadot-sdk#6398]
(Issue reserved for training purposes)

The `pallet-example-basic` should have all the pallet hooks as demonstration. As a start, the `on_idle` could be added.

**Comment by ggwpez** [comment]
Yes looks good. Please open a merge-request (if you are comfortable with this 😄).

**Comment by rainbow-promise** [comment]
Hey @annatekl-dev please signify if you still working on this. Thanks ☺️

**Comment by Krayt78** [comment]
@ggwpez  is it ok to take over this one since there hasnt been any replies since ?

**Comment by rainbow-promise** [comment]
> @ggwpez  is it ok to take over this one since there hasnt been any replies since ?
> 
> 

I'm currently working on it but quickest keyboard wins 😋

**Comment by Krayt78** [comment]
> > @ggwpez  is it ok to take over this one since there hasnt been any replies since ?
> 
> I'm currently working on it but quickest keyboard wins 😋

Nah that would be a waste of resources, if you are already on it i will leave it to you but i dont see any commit or branch on your side showing that you are curently working on it ^^'

**Comment by abdulwaarith0** [comment]
Hi @ggwpez — this looks abandoned: @annatekl-dev drafted the `on_idle` hook and you approved it back in Nov 2024, but no PR was ever opened, and the two later folks who said they'd take it over (@rainbow-promise / @Krayt78) also went quiet — no activity or linked PR for ~20 months.

I'd like to pick this up. Plan: extend the `#[pallet::hooks]` impl in `pallet-example-basic` with the currently-missing hooks as documented demonstrations — starting with `on_idle` as you suggested, plus `on_poll`, `on_runtime_upgrade`, `integrity_test`, and a feature-gated `try_state`, each with explanatory doc comments in the same teaching style as the existing `on_initialize`/`on_finalize`/`offchain_worker`. Happy to keep it to just `on_idle` first if you'd prefer a smaller PR. Is it OK for me to take it?
